include/bm/bm_sim/field_lists.h

```diff
@@ -46,23 +47,42 @@ class FieldList {
     bool operator==(const field_t& other) const {
       return header == other.header && offset == other.offset;
     }
-
     bool operator!=(const field_t& other) const {
       return !(*this == other);
     }
   };
 
+  struct constant_t {
+    int value;
+    size_t nbits;
+
+    bool operator==(const constant_t& other) const {
+      return value == other.value && nbits == other.nbits;
+    }
+    bool operator!=(const constant_t& other) const {
+      return !(*this == other);
+    }
+  };
+
  public:
-  using iterator = std::vector<field_t>::iterator;
-  using const_iterator = std::vector<field_t>::const_iterator;
-  using reference = std::vector<field_t>::reference;
-  using const_reference = std::vector<field_t>::const_reference;
+  using field_list_member_t = boost::variant<field_t, constant_t>;
```

Comment:
Okay. But why not just make the visit function take a const reference to FieldListVisitor?